OpenSim/Common/STOFileAdapter.h

```diff
@@ -27,28 +27,141 @@
 
 namespace OpenSim {
 
+class STODataTypeNotSupported : public Exception {
+public:
+    STODataTypeNotSupported(const std::string& file,
+                            size_t line,
+                            const std::string& func,
+                            const std::string& datatype) :
+        Exception(file, line, func) {
+        std::string msg = "Datatype '" + datatype + "' is not supported.";
+
+        addMessage(msg);
+    }
+};
+
+class STODataTypeNotFound : public Exception {
+public:
+    STODataTypeNotFound(const std::string& file,
+                        size_t line,
+                        const std::string& func) :
+        Exception(file, line, func) {
+        std::string msg = "DataType not specified in the header.";
+
+        addMessage(msg);
+    }
+};
+
 /** STOFileAdapter is a DelimFileAdapter that presets the delimiters 
-appropriately for STO files.                                                  */
-class OSIMCOMMON_API STOFileAdapter : public DelimFileAdapter {
+appropriately for STO files. The format of the file is as follows:
+\code
+.................................
+.................................
+........ <metadata> .............
+.................................
+.................................
+endheader
+time colname1 colname2 ..........
+.................................
+.................................
+............ <data-rows> ........
+.................................
+.................................
+\endcode
+For example a TimeSeriesTable_<SimTK::Vec3> could look like:
+\code
+units=m
+DataType=Vec3
+endheader
```

Comment:
The only metadata we introduce are DataType and Version. Rest are user controlled metadata. I don't think this is important enough to hold this PR.

The other issue is that not all metadata is serializable. When DataTable was first implemented, I was told there was no requirement for metadata to be serializable.